allennlp/data/dataset_readers/gqa.py

```diff
@@ -0,0 +1,161 @@
+    ) -> None:
+        super().__init__(
+            image_dir,
+            image_loader,
+            image_featurizer,
+            region_detector,
+            feature_cache_dir=feature_cache_dir,
+            tokenizer=tokenizer,
+            token_indexers=token_indexers,
+            cuda_device=cuda_device,
+            max_instances=max_instances,
+            image_processing_batch_size=image_processing_batch_size,
+            skip_image_feature_extraction=skip_image_feature_extraction,
+        )
+        self.data_dir = data_dir
+
+    @overrides
+    def _read(self, split_or_filename: str):
+
+        if not self.data_dir:
+            self.data_dir = "https://nlp.stanford.edu/data/gqa/questions1.2.zip!"
+
+        splits = {
+            "challenge_all": f"{self.data_dir}challenge_all_questions.json",
+            "challenge_balanced": f"{self.data_dir}challenge_balanced_questions.json",
+            "test_all": f"{self.data_dir}test_all_questions.json",
+            "test_balanced": f"{self.data_dir}test_balanced_questions.json",
+            "testdev_all": f"{self.data_dir}testdev_all_questions.json",
+            "testdev_balanced": f"{self.data_dir}testdev_balanced_questions.json",
+            "train_balanced": f"{self.data_dir}train_balanced_questions.json",
+            "train_all": f"{self.data_dir}train_all_questions",
+            "val_all": f"{self.data_dir}val_all_questions.json",
+            "val_balanced": f"{self.data_dir}val_balanced_questions.json",
+        }
+
+        filename = splits.get(split_or_filename, split_or_filename)
+
+        # If we're considering a directory of files (such as train_all)
+        # loop through each in file in generator
+        if os.path.isdir(filename):
+            files = [f"{filename}{file_path}" for file_path in os.listdir(filename)]
+        else:
+            files = [filename]
+
+        for data_file in files:
+            with open(cached_path(data_file, extract_archive=True)) as f:
+                questions_with_annotations = json.load(f)
+
+            # It would be much easier to just process one image at a time, but it's faster to process
+            # them in batches. So this code gathers up instances until it has enough to fill up a batch
+            # that needs processing, and then processes them all.
+            question_dicts = list(
+                self.shard_iterable(
+                    questions_with_annotations[q_id] for q_id in questions_with_annotations
+                )
+            )
+
+            processed_images = self._process_image_paths(
+                self.images[f"{question_dict['imageId']}.jpg"] for question_dict in question_dicts
+            )
+
+            for question_dict, processed_image in zip(question_dicts, processed_images):
+                answers = {
+                    "answer": question_dict["answer"],
+                    "fullAnswer": question_dict["fullAnswer"],
```

Comment:
We don't actually return `fullAnswer` in the `text_to_instance` function, right? Any reason to pass it to the function?

Reply:
Ahh I wasn't sure if fullAnswer would be used later but I'll remove it!